webapp/channels/src/components/advanced_text_editor/show_formatting/show_formatting.test.tsx

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {fireEvent, screen} from '@testing-library/react';
+import React from 'react';
+import {IntlProvider} from 'react-intl';
+
+import {renderWithContext} from 'tests/react_testing_utils';
+
+import ShowFormatting from './show_formatting';
+
+jest.mock('components/with_tooltip', () => {
+    return ({children}: { children: React.ReactNode }) => <div>{children}</div>;
+});
+
+describe('ShowFormatting Component', () => {
+    it('should render correctly with default props', () => {
+        renderWithContext(
+            <IntlProvider locale='en'>
+                <ShowFormatting
+                    onClick={jest.fn()}
+                    active={false}
+                />
+            </IntlProvider>,
+        );
+
+        expect(screen.getByLabelText('Eye Icon')).toBeInTheDocument();
+    });
+
+    it('should call onClick handler when clicked', () => {
+        const onClick = jest.fn();
+        renderWithContext(
+            <IntlProvider locale='en'>
```

Comment:
` <IntlProvider locale='en'>` is already wrapped in renderWithContext so we dont need this